net/netcheck/netcheck.go

```diff
@@ -1597,12 +1605,46 @@ func (c *Client) nodeAddr(ctx context.Context, n *tailcfg.DERPNode, proto probeP
 		return
 	}
 
-	// TODO(bradfitz): add singleflight+dnscache here.
-	addrs, _ := net.DefaultResolver.LookupIPAddr(ctx, n.HostName)
+	c.mu.Lock()
```

Comment:
lot of lines between this Lock and the Unlock in this otherwise unrelated method.  can any/all of this be pulled out into a new method?

Reply:
Moved the default resolver function out of the lock section; PTAL?